go-mode.el

```diff
@@ -575,6 +575,18 @@ current line will be returned."
       (skip-chars-forward "^}")
       (forward-char))))
 
+(defun go--add-log-current-defun ()
+  "Return the name of the current defun.
+This is to be used as `add-log-current-defun-function'."
+  ;; TODO this should also deal with types and maybe import.
+  (save-excursion
+    (beginning-of-defun)
```

Comment:
We should probably also check if we're in/looking at a function at all, to avoid bogus results when calling `go--add-log-current-defun` outside of a function. But that can go in a future change.